aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -122,9 +122,10 @@ std::vector<Tensor> where(const Tensor& condition) {
 }
 
 Tensor _s_where_cpu(const Tensor& condition, const Tensor& self, const Tensor& other) {
-  Tensor ret = at::empty(self.sizes(), self.options());
+  auto common_type = at::promote_types(self.scalar_type(), other.scalar_type());
```

Comment:
I believe this should be based on the result_type, not the promote_types.

Reply:
^ yes we want to promote (using result_type) based on the tensor parameters, not directly on the dtypes. the tensors may be wrapped_numbers or zero-dim which gets handled differently.

Reply:
Previously result type was calculating based on self dtype

Reply:
not sure what you mean by that. `at::tensor(self, other)` (from TypeProperties.cpp) accepts Tensors, promote_types only accepts dtypes.

Reply:
I see, make sense. I will change it to at::tensor(self, other) instead

Reply:
sorry I meant `result_type(self, other)`

Reply:
yes, I got it thanks